arangod/Aql/Functions.cpp

```diff
@@ -6163,8 +6161,10 @@ AqlValue functions::GeoEquals(ExpressionContext* expressionContext,
   AqlValueMaterializer mat2(vopts);
 
   geo::ShapeContainer first, second;
-  Result res1 = geo::geojson::parseRegion(mat1.slice(p1, true), first, false);
-  Result res2 = geo::geojson::parseRegion(mat2.slice(p2, true), second, false);
+  auto res1 = geo::json::parseRegion(mat1.slice(p1, true), first,
```

Comment:
I really believe user of some C++ functions never should write `Result` or `Result<T>` because it's inconvenient and oblivious.

But if you insist I can revert such changes if you point they.

Reply:
I do not insist, this is your style, so go for it. But here is why I do not share your believe: Typically, code is written once and read many times. For the writer it might be a tiny bit more convenient to write "auto" instead of "Result" or even "Result<T>" (after all, the writer potentially would have to check the signature of the function he calls, but the writer will have to do this anyway). But for the reader, it is suddenly obvious what the type of `res1` is, which is an enormous advantage when reading code. The reader does not have to check the signature of `geo::json::parseRegion` to find out what type `res1` is to judge if subsequent code is correct. Therefore I believe that it is well spent time for the writer to save time and effort for the many readers of this code.